cpp/simulations/abm.cpp

```diff
@@ -849,9 +846,14 @@ int main()
     mio::abm::close_social_events(t_lockdown, 0.9, world.get_migration_parameters());
 
     auto sim = mio::abm::Simulation(t0, std::move(world));
+    return sim;
+}
 
-    sim.advance(tmax);
-
+/**
+ * Save the current run result to a directory.
+ * @param results the result in form of a TimeSeries object of infection_state individuals for the whole world
+*/
+void save_result(mio::TimeSeries<double> results, size_t run_idx, const fs::path& result_dir) {
```

Comment:
As a first output, this is certainly fine. We should consider putting this functionality to a different file to access it from anywhere and can easily add more options, as is done in the ODE. Once the world is copyable, there should also be an option to run a simulation multiple times with a parameter.

Reply:
I agree since the file is now too long to read. Maybe it can be put in a header with also the input parameters/variables so we have control for them.

Reply:
I've now switched to use save_result from result_io.h. Now we just need to store location population data in ABM TimeSeries object and update simulation output (#465)